### PR TITLE
Enable support of multiple exprtk adaptors (helpful if G+Smo is used in header-only mode)

### PR DESCRIPTION
**Pull request by mmoelle1**
This commit makes it possible to use multiple export adaptors at the same time.
This is particularly helpful of G+Smo is used as header-only library, when one might want to use the standard adaptor and mix them with adaptors for GMP/MPFR.

This commit is a preparation for committing support for Universal NUMbers (UNUMs).

### CODE REVIEW

File: src/gsCore/gsFunctionExpr.hpp [hunk 1]
```diff
@@ -52,23 +52,31 @@
...
+#include <exprtk_mpfr_adaptor.hpp> // external file
+#endif
+
+#if defined(GISMO_WITH_MPQ)
+#include <exprtk_gmp_adaptor.hpp>  // external file
 #endif
 
+#if defined(GISMO_WITH_CODIPACK)
+#include <gsCoDiPack/exprtk_codi_rf_adaptor.hpp>
+#include <gsCoDiPack/exprtk_codi_rr_adaptor.hpp>
```
Comment (filiatra) [inline src/gsCore/gsFunctionExpr.hpp]: double iinclusion!

Comment (filiatra) [reply src/gsCore/gsFunctionExpr.hpp]: ok, I did not see the rr/rf :)

**Comment by filiatra** [comment]
Seems that constants are always the same (doubles) , how about moving it to a file exprtk_constants.h  ?

**Comment by mmoelle1** [comment]
No, they are often of type double but, GMP/MPFR (and later Unum) use different types.

**Comment by filiatra** [comment]
As far as I can see GMP used double, maybe also for Unum is enough ? I dont know!

**Pull request merged**